backend/dataall/cdkproxy/stacks/environment.py

```diff
@@ -162,19 +159,91 @@ def __init__(self, scope, id, target_uri: str = None, **kwargs):
-            models.EnvironmentGroup
-        ] = self.get_environment_groups(self.engine, environment=self._environment)
+        self.environment_groups: [models.EnvironmentGroup] = self.get_environment_groups(self.engine, environment=self._environment)
 
-        self.environment_admins_group: [
-            models.EnvironmentGroup
-        ] = self.get_environment_admins_group(self.engine, self._environment)
+        self.environment_admins_group: models.EnvironmentGroup = self.get_environment_admins_group(self.engine, self._environment)
 
         self.all_environment_datasets = self.get_all_environment_datasets(
             self.engine, self._environment
         )
 
-        self.check_sagemaker_studio(engine=self.engine, environment=self._environment)
+        self.sagemaker_domain_exists = self.check_sagemaker_studio(engine=self.engine, environment=self._environment)
+
+        if self._environment.mlStudiosEnabled and not(self.sagemaker_domain_exists):
+
+            sagemaker_domain_role = iam.Role(
+                self,
+                'RoleForSagemakerStudioUsers',
+                assumed_by=iam.ServicePrincipal('sagemaker.amazonaws.com'),
+                role_name="RoleSagemakerStudioUsers",
+                managed_policies=[iam.ManagedPolicy.from_managed_policy_arn(
+                    self,
+                    id="SagemakerFullAccess",
+                    managed_policy_arn="arn:aws:iam::aws:policy/AmazonSageMakerFullAccess"),
+                    iam.ManagedPolicy.from_managed_policy_arn(
+                    self,
+                        id="S3FullAccess",
+                        managed_policy_arn="arn:aws:iam::aws:policy/AmazonS3FullAccess")
+                ]
+            )
+
+            sagemaker_domain_key = kms.Key(
+                self,
+                'SagemakerDomainKmsKey',
+                alias="SagemakerStudioDomain",
+                enable_key_rotation=True,
+                policy=iam.PolicyDocument(
+                    assign_sids=True,
+                    statements=[
+                        iam.PolicyStatement(
+                            resources=['*'],
+                            effect=iam.Effect.ALLOW,
+                            principals=[
+                                iam.AccountPrincipal(account_id=self._environment.AwsAccountId),
+                                iam.Role.from_role_arn(self, 'DomainRole', role_arn=sagemaker_domain_role.role_arn),
+                                iam.Role.from_role_arn(self, 'EnvironmentDefaultRole', role_arn=self.environment_admins_group.environmentIAMRoleArn),
+                            ] + [iam.Role.from_role_arn(self, f'{group.groupUri}Role', role_arn=group.environmentIAMRoleArn) for group in self.environment_groups],
+                            actions=['kms:*'],
+                        )
+                    ],
+                ),
+            )
+
+            default_vpc = ec2.Vpc.from_lookup(self, 'VPCStudio', is_default=True)
```

Comment:
What is going to happen in case of no default VPC in the AWS account?

Reply:
the PR assumes the VPC as input:
_"This takes as an input your default VPC, and assumes it has the appropriate networking configurations. To learn about the network requirements for Sagemaker Studio, read this documentation https://docs.aws.amazon.com/sagemaker/latest/dg/studio-notebooks-and-internet-access.html. For advanced networking customizations, stacks/environment.py should be modified according to your ecosystem's needs."_

Reply:
yeah but here in line 212  is see a lookup for the default vpc but there will be cases where the account will not have the default vpc at all so the question is how it is going to work? Shouldn't it look more for the vpc when the data.all is being deployed (vpcId passed as an input or the one which was created as a part of deploying the data.all) or am I missing something?

Reply:
ah ok I see, so it simply assumes the environment which we link in data.all is having a default vpc

Reply:
By looking at the code, you can also choose to create a SageMaker Studio domain before creating the environment and avoid data.all to create it for you

Reply:
@JahedZ  Here a suggestion:

```suggestion
try:
      default_vpc = ec2.Vpc.from_lookup(self, 'VPCStudio', is_default=True)
except Exception as e:
       logger.error(f"Default VPC not found, Exception: {e}")
```
```

Reply:
Indeed, this is thought as the "minimal network setup" to have a viable Studio environment - provided that you don't already own a Studio domain. Thanks for the suggestion, it's a very good addition

Reply:
But it would be interested to have some kind of "Bring your own VPC" Setup, upon workspace creation.
Similar to what we do with Sagemaker notebook instances